cardano/onchain/lib/ibc/client/ics-007-tendermint-client/cometbft/block/block_id.ak

```diff
@@ -16,16 +16,9 @@ pub fn null_block_id() -> BlockID {
 }
 
 pub fn validate_basic(block_id: BlockID) -> Bool {
-  expect validate_hash(block_id.hash)
-
-  expect validate_hash(block_id.part_set_header.hash)
-
-  expect block_id.part_set_header.total >= 0
-
-  True
-}
-
-fn validate_hash(h: ByteArray) -> Bool {
-  let len = h |> bytearray.length()
-  len == 0 || len == tm_hash_size
+  and {
+    is_empty_or_size(block_id.hash, tm_hash_size),
+    is_empty_or_size(block_id.part_set_header.hash, tm_hash_size),
+    block_id.part_set_header.total >= 0,
+  }
```

Comment:
note for reviewer, `expect` is cool and all, but it's expensive and at times unnecessary. `and{}` works just fine in this case.

expect feels more appropriate when asserting against a deconstruction.